funsor/gaussian.py

```diff
@@ -528,6 +530,11 @@ def unscaled_sample(self, sampled_vars, sample_inputs):
         raise NotImplementedError('TODO implement partial sampling of real variables')
 
 
+@substitute.register(Gaussian, dict)
+def subs_gaussian(expr, subs):
+    return expr.eager_subs(tuple((k, to_funsor(v, expr.inputs[k]) if k in expr.inputs else v) for k, v in subs.items()))
```

Comment:
Unlike `Joint`, `Gaussian` does introduce fresh variables, so calling `eager_subs` is fine here.